### PR TITLE
quicksynergy: init at 0.9.0

### PR DESCRIPTION
**Pull request by spinus**

### CODE REVIEW

**Comment by vcunat** [comment]
- Passing whole `pkgs` is considered a bad style IIRC.
- Plugging into all-packages.nix is missing.
- I'm suspicious of `license = pkgs.license.GPL2;` (existence of this attribute path).

**Comment by spinus** [comment]
@vcunat : thank you for review.
1. should I do ``with pkgs`` at the beginning? (I thought explicit better than implicit, just to understand where stuff is comming from, but maybe it's just a beginner issue)
2. Sure, will add.
3. I'll check and fix.

**Comment by spinus** [comment]
@vcunat : better now?

File: pkgs/applications/misc/quicksynergy/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{stdenv, fetchurl, pkgs}:
```
Comment (pSub) [inline pkgs/applications/misc/quicksynergy/default.nix]: Explicitly pass the dependencies here. I.e. replace `pkgs` with `pkgsconfig, gtk2, synergy`. You can basically look at any other package for examples.

**Comment by jgillich** [comment]
Also I think `version` should be part of the derivation.

**Comment by vcunat** [comment]
> Also I think version should be part of the derivation.

That certainly isn't required.

File: pkgs/applications/misc/quicksynergy/default.nix (hunk 1 above)
Comment (vcunat) [inline pkgs/applications/misc/quicksynergy/default.nix]: Yeah, that's exactly what I meant.

**Comment by spinus** [comment]
OK, will change that.

few questions.
1. Is there any mechanism to throw an error when two the same names are "imported/inherited" to the namespace?
2. Is there any way to trace where the variable was imported from? (except grep :)

**Comment by vcunat** [comment]
- 1: you can't have multiple bindings of the same name within the same attribute map (for example), but there is the usual concept of inner-scope bindings shadowing outer-scope bindings.
- 2: maybe nothing better than `builtins.unsafeGetAttrPos`.

**Comment by spinus** [comment]
1. Oh ok, so I was little sceptical about this shadowing stuff (I wasn't sure if it exsits or not :-)
2. Thank you, will check that.

File: pkgs/applications/misc/quicksynergy/default.nix (hunk 1 above)
Comment (spinus) [inline pkgs/applications/misc/quicksynergy/default.nix]: btw, so how do you guys know, which argument is "pkgs" and which is nix function (let's say, you are new person and don't know, grepping is the best option here? any other way to find what is what?)

**Comment by spinus** [comment]
@jgillich, @vcunat (version) I would love that, but that's not only not required, but even impossible right now I think:
```
error: undefined variable ‘version’ at
```

**Comment by spinus** [comment]
updated

File: pkgs/applications/misc/quicksynergy/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{stdenv, fetchurl, pkgconfig, gtk2, synergy}:
+let
+  version = "0.9.0";
+in
+stdenv.mkDerivation {
```
Comment (globin) [inline pkgs/applications/misc/quicksynergy/default.nix]: change this to `stdenv.mkDerivation rec {` and move the version into the derivation

**Comment by vcunat** [comment]
I pushed to master and 15.09, with that nitpick fixed.

File: pkgs/applications/misc/quicksynergy/default.nix (hunk 2 above)
Comment (spinus) [inline pkgs/applications/misc/quicksynergy/default.nix]: Good tip, thank you.

btw, why inherit is used in the second case, is it required or without 'inherit' the case would be sufficient?

**Comment by spinus** [comment]
@vcunat thank you.

File: pkgs/applications/misc/quicksynergy/default.nix (hunk 2 above)
Comment (globin) [inline pkgs/applications/misc/quicksynergy/default.nix]: Inherit  pulls the `attr` into the hash itself making `hash.attr` work.

**Pull request closed without merging**